### PR TITLE
OKTA-592141 Updates to Event Hook Mgmt API for monthly

### PR DESCRIPTION
**Pull request by brianduffield-okta**
## Description:
- **What's changed?** Added new Event Hook filter parameters to the API
- **Is this PR related to a Monolith release?** Removed from 2023.04.0 release. Release TBD

### Resolves:

* [OKTA-592141](https://oktainc.atlassian.net/browse/OKTA-592141)

### CODE REVIEW

**Comment by acrolinxdevdocs-okta** [comment]
## Acrolinx score **DRAFT 06.01.22**
 
A minimum Acrolinx Score of **40** is required. The total score is an average of the subscores.

Select **Total score** to review the Acrolinx scorecard for your article. Try to increase your individual scores, ex. Correctness. Your article will be clearer and more consistent with Microsoft standards.
 
| Article | **Total score**<br />**Required:70** | Word and phrases<br />(Brand, terms)<br />Preferred: 40 | Correctness<br />(Spelling, grammar)<br />Preferred: 40 | Clarity<br />(Readability)<br />Preferred: 10 | Inclusive language<br />(+ accesibility)<br />Preferred: 10 |
|---------|:------------------------------------:|:--------------------------------------------------------:|:--------------------------------------------------------:|:---------------------------------------------:|:------------------------------------------------------------:|
| [:white_check_mark:](# "Successfully processed") [packages/@okta/vuepress-site/docs/reference/api/event-hooks/index.md](https://github.com/okta/okta-developer-docs/blob/c35b268deb8a35ed803c73ecdb7304280508cdb0/packages%2F%40okta%2Fvuepress-site%2Fdocs%2Freference%2Fapi%2Fevent-hooks%2Findex.md) | [64](https://okta.acrolinx.cloud/api/v1/checking/scorecards/d76fb907-f7eb-472f-9182-085bc37ffaa8) | 56  | 70  | 42  | 73  |

  - The results from this response were procced by the [hugo-targets1.solution.acrolinx.sh](https://hugo-targets1.solution.acrolinx.sh) Acrolinx Cloud Host.
  - Scorecard links by default are only good for 7 days. [Link for this commit's Content Analysis Dashboard](https://okta.acrolinx.cloud/api/batch/gen.github.664103a8-5718-4e9c-a022-a6440ccefb05?csBaseUrl=https%3A%2F%2Fokta.acrolinx.cloud&X-Acrolinx-Client-Locale=en)

**Comment by acrolinxdevdocs-okta** [comment]
## Acrolinx score **DRAFT 06.01.22**
 
A minimum Acrolinx Score of **40** is required. The total score is an average of the subscores.

Select **Total score** to review the Acrolinx scorecard for your article. Try to increase your individual scores, ex. Correctness. Your article will be clearer and more consistent with Microsoft standards.
 
| Article | **Total score**<br />**Required:70** | Word and phrases<br />(Brand, terms)<br />Preferred: 40 | Correctness<br />(Spelling, grammar)<br />Preferred: 40 | Clarity<br />(Readability)<br />Preferred: 10 | Inclusive language<br />(+ accesibility)<br />Preferred: 10 |
|---------|:------------------------------------:|:--------------------------------------------------------:|:--------------------------------------------------------:|:---------------------------------------------:|:------------------------------------------------------------:|
| [:white_check_mark:](# "Successfully processed") [packages/@okta/vuepress-site/docs/reference/api/event-hooks/index.md](https://github.com/okta/okta-developer-docs/blob/868f5cfeed328e438ac2172255ed69f26847b670/packages%2F%40okta%2Fvuepress-site%2Fdocs%2Freference%2Fapi%2Fevent-hooks%2Findex.md) | [65](https://okta.acrolinx.cloud/api/v1/checking/scorecards/00ad8450-ca55-4237-bd45-c5b8e9650123) | 56  | 76  | 42  | 73  |

  - The results from this response were procced by the [hugo-targets1.solution.acrolinx.sh](https://hugo-targets1.solution.acrolinx.sh) Acrolinx Cloud Host.
  - Scorecard links by default are only good for 7 days. [Link for this commit's Content Analysis Dashboard](https://okta.acrolinx.cloud/api/batch/gen.github.d0d6aa94-c2a5-4012-b564-f0ac4d0a48a0?csBaseUrl=https%3A%2F%2Fokta.acrolinx.cloud&X-Acrolinx-Client-Locale=en)

**Comment by acrolinxdevdocs-okta** [comment]
## Acrolinx score **DRAFT 06.01.22**
 
A minimum Acrolinx Score of **40** is required. The total score is an average of the subscores.

Select **Total score** to review the Acrolinx scorecard for your article. Try to increase your individual scores, ex. Correctness. Your article will be clearer and more consistent with Microsoft standards.
 
| Article | **Total score**<br />**Required:70** | Word and phrases<br />(Brand, terms)<br />Preferred: 40 | Correctness<br />(Spelling, grammar)<br />Preferred: 40 | Clarity<br />(Readability)<br />Preferred: 10 | Inclusive language<br />(+ accesibility)<br />Preferred: 10 |
|---------|:------------------------------------:|:--------------------------------------------------------:|:--------------------------------------------------------:|:---------------------------------------------:|:------------------------------------------------------------:|
| [:white_check_mark:](# "Successfully processed") [packages/@okta/vuepress-site/docs/reference/api/event-hooks/index.md](https://github.com/okta/okta-developer-docs/blob/41b70605b3b039b6ed43d80d2988d5b34ab1f101/packages%2F%40okta%2Fvuepress-site%2Fdocs%2Freference%2Fapi%2Fevent-hooks%2Findex.md) | [65](https://okta.acrolinx.cloud/api/v1/checking/scorecards/ba29669f-9666-40a6-815e-d41080edccfb) | 56  | 76  | 42  | 73  |

  - The results from this response were procced by the [hugo-targets1.solution.acrolinx.sh](https://hugo-targets1.solution.acrolinx.sh) Acrolinx Cloud Host.
  - Scorecard links by default are only good for 7 days. [Link for this commit's Content Analysis Dashboard](https://okta.acrolinx.cloud/api/batch/gen.github.862ebbee-9862-4a3d-bdfd-0f04a34ab703?csBaseUrl=https%3A%2F%2Fokta.acrolinx.cloud&X-Acrolinx-Client-Locale=en)

**Comment by acrolinxdevdocs-okta** [comment]
## Acrolinx score **DRAFT 06.01.22**
 
A minimum Acrolinx Score of **40** is required. The total score is an average of the subscores.

Select **Total score** to review the Acrolinx scorecard for your article. Try to increase your individual scores, ex. Correctness. Your article will be clearer and more consistent with Microsoft standards.
 
| Article | **Total score**<br />**Required:70** | Word and phrases<br />(Brand, terms)<br />Preferred: 40 | Correctness<br />(Spelling, grammar)<br />Preferred: 40 | Clarity<br />(Readability)<br />Preferred: 10 | Inclusive language<br />(+ accesibility)<br />Preferred: 10 |
|---------|:------------------------------------:|:--------------------------------------------------------:|:--------------------------------------------------------:|:---------------------------------------------:|:------------------------------------------------------------:|
| [:white_check_mark:](# "Successfully processed") [packages/@okta/vuepress-site/docs/reference/api/event-hooks/index.md](https://github.com/okta/okta-developer-docs/blob/7ba3e0bc437e3093aa441006ef1f73b36c96a347/packages%2F%40okta%2Fvuepress-site%2Fdocs%2Freference%2Fapi%2Fevent-hooks%2Findex.md) | [61](https://okta.acrolinx.cloud/api/v1/checking/scorecards/5d34b3b7-6eeb-4756-8edc-fd61347dc5b3) | 54  | 56  | 58  | 73  |

  - The results from this response were procced by the [hugo-targets1.solution.acrolinx.sh](https://hugo-targets1.solution.acrolinx.sh) Acrolinx Cloud Host.
  - Scorecard links by default are only good for 7 days. [Link for this commit's Content Analysis Dashboard](https://okta.acrolinx.cloud/api/batch/gen.github.bf10b0ea-b593-46b8-8781-7e05c3dad160?csBaseUrl=https%3A%2F%2Fokta.acrolinx.cloud&X-Acrolinx-Client-Locale=en)

**Comment by acrolinxdevdocs-okta** [comment]
## Acrolinx score **DRAFT 06.01.22**
 
A minimum Acrolinx Score of **40** is required. The total score is an average of the subscores.

Select **Total score** to review the Acrolinx scorecard for your article. Try to increase your individual scores, ex. Correctness. Your article will be clearer and more consistent with Microsoft standards.
 
| Article | **Total score**<br />**Required:70** | Word and phrases<br />(Brand, terms)<br />Preferred: 40 | Correctness<br />(Spelling, grammar)<br />Preferred: 40 | Clarity<br />(Readability)<br />Preferred: 10 | Inclusive language<br />(+ accesibility)<br />Preferred: 10 |
|---------|:------------------------------------:|:--------------------------------------------------------:|:--------------------------------------------------------:|:---------------------------------------------:|:------------------------------------------------------------:|
| [:white_check_mark:](# "Successfully processed") [packages/@okta/vuepress-site/docs/reference/api/event-hooks/index.md](https://github.com/okta/okta-developer-docs/blob/dbeba2842debce271925d30eb0cff49e88634cf9/packages%2F%40okta%2Fvuepress-site%2Fdocs%2Freference%2Fapi%2Fevent-hooks%2Findex.md) | [61](https://okta.acrolinx.cloud/api/v1/checking/scorecards/cb7c9575-690b-49a7-950c-d51028baaa4d) | 54  | 56  | 58  | 73  |

  - The results from this response were procced by the [hugo-targets1.solution.acrolinx.sh](https://hugo-targets1.solution.acrolinx.sh) Acrolinx Cloud Host.
  - Scorecard links by default are only good for 7 days. [Link for this commit's Content Analysis Dashboard](https://okta.acrolinx.cloud/api/batch/gen.github.89b57fc8-6710-4111-8bab-8fc5109cf9ee?csBaseUrl=https%3A%2F%2Fokta.acrolinx.cloud&X-Acrolinx-Client-Locale=en)

**Comment by acrolinxdevdocs-okta** [comment]
## Acrolinx score **DRAFT 06.01.22**
 
A minimum Acrolinx Score of **40** is required. The total score is an average of the subscores.

Select **Total score** to review the Acrolinx scorecard for your article. Try to increase your individual scores, ex. Correctness. Your article will be clearer and more consistent with Microsoft standards.
 
| Article | **Total score**<br />**Required:70** | Word and phrases<br />(Brand, terms)<br />Preferred: 40 | Correctness<br />(Spelling, grammar)<br />Preferred: 40 | Clarity<br />(Readability)<br />Preferred: 10 | Inclusive language<br />(+ accesibility)<br />Preferred: 10 |
|---------|:------------------------------------:|:--------------------------------------------------------:|:--------------------------------------------------------:|:---------------------------------------------:|:------------------------------------------------------------:|
| [:white_check_mark:](# "Successfully processed") [packages/@okta/vuepress-site/docs/reference/api/event-hooks/index.md](https://github.com/okta/okta-developer-docs/blob/48a8cf6cd45729b2f2028304d234d512b31c858b/packages%2F%40okta%2Fvuepress-site%2Fdocs%2Freference%2Fapi%2Fevent-hooks%2Findex.md) | [61](https://okta.acrolinx.cloud/api/v1/checking/scorecards/461d6f89-ce6a-41a5-8aba-fc23eb346280) | 54  | 54  | 58  | 73  |

  - The results from this response were procced by the [hugo-targets1.solution.acrolinx.sh](https://hugo-targets1.solution.acrolinx.sh) Acrolinx Cloud Host.
  - Scorecard links by default are only good for 7 days. [Link for this commit's Content Analysis Dashboard](https://okta.acrolinx.cloud/api/batch/gen.github.eee554a7-63d0-42de-a63f-10eaaf41fbda?csBaseUrl=https%3A%2F%2Fokta.acrolinx.cloud&X-Acrolinx-Client-Locale=en)

File: packages/@okta/vuepress-site/docs/reference/api/event-hooks/index.md [hunk 1]
```diff
@@ -628,6 +637,59 @@ To use Basic Auth, set `type` to `HEADER`, `key` to `Authorization`, and `value`
...
+| Property | Description                                                                  | DataType | Required | ReadOnly |
+|----------|------------------------------------------------------------------------------|----------|----------|----------|
+| event    | The filtered event type   | String   | TRUE     | FALSE   |
+| condition    | The object that defines the filter using Okta Expression Language      | [condition object](#condition-object)  | TRUE    | FALSE    |
+
+#### Condition object
+
+| Property | Description                                                                  | DataType | Required | ReadOnly |
+|----------|------------------------------------------------------------------------------|----------|----------|----------|
+| version    | Internal field   | String   | TRUE     | TRUE   |
```
Comment (isaacdurham-okta) [inline packages/@okta/vuepress-site/docs/reference/api/event-hooks/index.md:691]: Does Internal field need more context?

File: packages/@okta/vuepress-site/docs/reference/api/event-hooks/index.md [hunk 2]
```diff
@@ -628,6 +637,59 @@ To use Basic Auth, set `type` to `HEADER`, `key` to `Authorization`, and `value`
...
+<ApiLifecycle access="ea" />
+
+<EventHookEANote/>
+
+Explore the event hooks API with filters: [![Run in Postman](https://run.pstmn.io/button.svg)](https://app.getpostman.com/run-collection/15901964-61517094-f0ed-45e9-982a-f7a6a2db1bab?action=collection%2Ffork&collection-url=entityId%3D15901964-61517094-f0ed-45e9-982a-f7a6a2db1bab%26entityType%3Dcollection%26workspaceId%3D9f1d6c8f-d027-4107-a5c5-20d963c2c9d8)
+
+| Property | Description                                                                  | DataType | Required | ReadOnly |
+|----------|------------------------------------------------------------------------------|----------|----------|----------|
+| event    | The filtered event type   | String   | TRUE     | FALSE   |
+| condition    | The object that defines the filter using Okta Expression Language      | [condition object](#condition-object)  | TRUE    | FALSE    |
```
Comment (isaacdurham-okta) [inline packages/@okta/vuepress-site/docs/reference/api/event-hooks/index.md:685]: Would this be simpler as just something like "The object that defines the filter"?

The Condition object mentions OEL already.

File: packages/@okta/vuepress-site/docs/reference/api/event-hooks/index.md [hunk 3]
```diff
@@ -97,7 +98,8 @@ curl -v -X POST \
         "items" : [
             "user.lifecycle.create",
             "user.lifecycle.activate"
-        ]
+        ],
+        "filter": null
```
Comment (isaacdurham-okta) [inline packages/@okta/vuepress-site/docs/reference/api/event-hooks/index.md:102]: Feel free to ignore this, but there are some spacing inconsistencies between colons, and differing indents for some of the "Filter" params.

**Review by isaacdurham-okta — APPROVED** [review]

File: packages/@okta/vuepress-site/docs/reference/api/event-hooks/index.md (hunk 1 above)
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/reference/api/event-hooks/index.md:691]: Nothing that would help customers.

File: packages/@okta/vuepress-site/docs/reference/api/event-hooks/index.md (hunk 3 above)
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/reference/api/event-hooks/index.md:102]: Thanks! Good eye.

**Comment by acrolinxdevdocs-okta** [comment]
## Acrolinx score **DRAFT 06.01.22**
 
A minimum Acrolinx Score of **40** is required. The total score is an average of the subscores.

Select **Total score** to review the Acrolinx scorecard for your article. Try to increase your individual scores, ex. Correctness. Your article will be clearer and more consistent with Microsoft standards.
 
| Article | **Total score**<br />**Required:70** | Word and phrases<br />(Brand, terms)<br />Preferred: 40 | Correctness<br />(Spelling, grammar)<br />Preferred: 40 | Clarity<br />(Readability)<br />Preferred: 10 | Inclusive language<br />(+ accesibility)<br />Preferred: 10 |
|---------|:------------------------------------:|:--------------------------------------------------------:|:--------------------------------------------------------:|:---------------------------------------------:|:------------------------------------------------------------:|
| [:white_check_mark:](# "Successfully processed") [packages/@okta/vuepress-site/docs/reference/api/event-hooks/index.md](https://github.com/okta/okta-developer-docs/blob/1e703037a00ee4740085b40ac40656a7ea3cf3b1/packages%2F%40okta%2Fvuepress-site%2Fdocs%2Freference%2Fapi%2Fevent-hooks%2Findex.md) | [62](https://okta.acrolinx.cloud/api/v1/checking/scorecards/ddd6b3e7-8aed-4fcc-8915-094c6f0c40ba) | 54  | 56  | 58  | 73  |

  - The results from this response were procced by the [hugo-targets1.solution.acrolinx.sh](https://hugo-targets1.solution.acrolinx.sh) Acrolinx Cloud Host.
  - Scorecard links by default are only good for 7 days. [Link for this commit's Content Analysis Dashboard](https://okta.acrolinx.cloud/api/batch/gen.github.3709f84e-3a4f-4c30-a351-afc27f4504e8?csBaseUrl=https%3A%2F%2Fokta.acrolinx.cloud&X-Acrolinx-Client-Locale=en)

**Pull request merged**